addons/mail/static/src/components/message/message_tests.js

```diff
@@ -983,6 +983,87 @@ QUnit.test('open chat with author on avatar click should be disabled when curren
     );
 });
 
+QUnit.test('Notification Canceled', async function (assert) {
+    assert.expect(8);
+
+    this.data['mail.channel'].records.push({ id: 11 });
+    this.data['res.partner'].records.push({ id: 12, name: "Someone" });
+    this.data['mail.message'].records.push({
+        id: 10,
+        channel_ids: [11],
+        message_type: 'email',
+        model: 'mail.channel',
+        notification_ids: [ 11 ],
+        res_id: 11,
+    });
+    this.data['mail.notification'].records.push({
+        id: 11,
+        failure_type: 'SMTP',
+        mail_message_id: 10,
+        notification_status: 'canceled',
+        notification_type: 'email',
+        res_partner_id: [ 12, "Someone" ],
+    });
+    await this.start();
+    const threadViewer = await this.env.models['mail.thread_viewer'].create({
+        hasThreadView: true,
+        thread: [['insert', {
+            id: 11,
+            model: 'mail.channel',
+        }]],
+    });
+    await this.env.models['mail.message'].performRpcMessageFetch([]);
+    const message = this.env.models['mail.message'].find(message => message.id === 10);
+    await this.createMessageComponent(message, {
+        threadViewLocalId: threadViewer.threadView.localId
+    });
+
+    assert.containsOnce(
+        document.body,
+        '.o_Message',
+        "should display a message component"
+    );
+    // await new Promise(a=>{});
+    assert.containsOnce(
+        document.body,
+        '.o_Message_notificationIconContainer',
+        "should display the notification icon container"
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_Message_notificationIcon',
+        "should display the notification icon"
+    );
+    assert.hasClass(
+        document.querySelector('.o_Message_notificationIcon'),
+        'fa-envelope-o',
+        "icon should represent email success"
```

Comment:
You're testing "canceled" and you assert success, this is very strange

Reply:
in fact it's the same icon, I'll just remove the word success